packages/host/tests/integration/components/operator-mode-test.gts

```diff
@@ -0,0 +1,177 @@
+import { Realm } from '@cardstack/runtime-common/realm';
+import { Loader } from '@cardstack/runtime-common/loader';
+import OperatorMode from '@cardstack/host/components/operator-mode';
+import CardPrerender from '@cardstack/host/components/card-prerender';
+import { Card } from 'https://cardstack.com/base/card-api';
+import { renderComponent } from '../../helpers/render-component';
+import { testRealmURL, setupCardLogs, setupMockLocalRealm, TestRealmAdapter, TestRealm } from '../../helpers';
+import { waitFor, click, fillIn } from '@ember/test-helpers';
+import type LoaderService from '@cardstack/host/services/loader-service';
+import { shimExternals } from '@cardstack/host/lib/externals';
+
+let cardApi: typeof import('https://cardstack.com/base/card-api');
+
+module('Integration | operator-mode', function (hooks) {
+  let adapter: TestRealmAdapter;
+  let realm: Realm;
+  setupRenderingTest(hooks);
+  setupMockLocalRealm(hooks);
+  setupCardLogs(
+    hooks,
+    async () => await Loader.import(`${baseRealm.url}card-api`)
+  );
+
+  async function loadCard(url: string): Promise<Card> {
+    let { createFromSerialized, recompute } = cardApi;
+    let result = await realm.searchIndex.card(new URL(url));
+    if (!result || result.type === 'error') {
+      throw new Error(
+        `cannot get instance ${url} from the index: ${
+          result ? result.error.detail : 'not found'
+        }`
+      );
+    }
+    let card = await createFromSerialized<typeof Card>(
+      result.doc.data,
+      result.doc,
+      undefined,
+      {
+        loader: Loader.getLoaderFor(createFromSerialized),
+      }
+    );
+    await recompute(card, { loadFields: true });
+    return card;
+  }
+
+  hooks.beforeEach(async function () {
+    Loader.addURLMapping(
+      new URL(baseRealm.url),
+      new URL('http://localhost:4201/base/')
+    );
+    shimExternals();
+    let loader = (this.owner.lookup('service:loader-service') as LoaderService)
+      .loader;
+    cardApi = await loader.import(`${baseRealm.url}card-api`);
+
+    adapter = new TestRealmAdapter({
+      'pet.gts': `
+        import { contains, field, Component, Card } from "https://cardstack.com/base/card-api";
+        import StringCard from "https://cardstack.com/base/string";
+
+        export class Pet extends Card {
+          @field name = contains(StringCard);
+          static embedded = class Embedded extends Component<typeof this> {
+            <template>
+              <h3 data-test-pet={{@model.name}}>
+                <@fields.name/>
+              </h3>
+            </template>
+          }
+        }
+      `,
+      'person.gts': `
+        import { contains, linksTo, field, Component, Card } from "https://cardstack.com/base/card-api";
+        import StringCard from "https://cardstack.com/base/string";
+        import { Pet } from "./pet";
+
+        export class Person extends Card {
+          @field firstName = contains(StringCard);
```

Comment:
This test should probably include a computed in order to verify that we handle them properly.